src/Workspaces/Core/Portable/Workspace/Solution/Checksum_Factory.cs

```diff
@@ -21,135 +19,41 @@ namespace Microsoft.CodeAnalysis
     internal partial record class Checksum
     {
         // https://github.com/dotnet/runtime/blob/f2db6d6093c54e5eeb9db2d8dcbe15b2db92ad8c/src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/SHA256.cs#L18-L19
-        private const int SHA256HashSizeBytes = 256 / 8;
+        private const int XXHash128SizeBytes = 128 / 8;
```

Comment:
not sure why we need this constant  *and* the HashSize constant.  But was keeping this roughly the same.

Reply:
Could you follow up with cleanup-refactoring? 
Other than having this duplicated I'm not sure about the value of having the Create methods in a separate file.

Reply:
yes.  i'm also intending to measure why we need Checksum *and* HashData when it seems like we could just have the latter.  The former is technically good if we expect a few actual distinct checksums, and tons of pointers to them.  The latter would be good if we expect less pointers and most of the values to just be needed in the place where we're pointing at them.

Reply:
We might have been a 32-bit process the last time this was evaluated, but the concern last time we looked was the number of pointers could exceed the number of instances.